sdk/src/test/java/com/stormpath/sdk/StormpathConfigurationTest.java

```diff
@@ -0,0 +1,45 @@
+package com.stormpath.sdk;
+
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class StormpathConfigurationTest extends BaseTest {
+
+    @Test(expected = IllegalStateException.class)
+    public void exceptionThrownIfBaseUrlNotSet() throws Exception {
+        new StormpathConfiguration.Builder().build();
+    }
+
+    @Test
+    public void pathsStartWithSlash() throws Exception {
+        StormpathConfiguration stormpath = new StormpathConfiguration.Builder()
+                .baseUrl("https://api.stormpath.com/v1")
+                .oauthPath("my-oauth")
+                .registerPath("my-register")
+                .passwordResetPath("my-password-reset")
+                .verifyEmailPath("my-verify-email")
+                .userProfilePath("my-me-path")
+                .logoutPath("my-logout-path")
+                .build();
+
+        assertThat(stormpath.oauthPath()).isEqualTo("/my-oauth");
+        assertThat(stormpath.registerPath()).isEqualTo("/my-register");
+        assertThat(stormpath.passwordResetPath()).isEqualTo("/my-password-reset");
+        assertThat(stormpath.verifyEmailPath()).isEqualTo("/my-verify-email");
+        assertThat(stormpath.userProfilePath()).isEqualTo("/my-me-path");
+        assertThat(stormpath.logoutPath()).isEqualTo("/my-logout-path");
+    }
+
+    @Test
+    public void baseUrlWithoutTrailingSlash() throws Exception {
+        StormpathConfiguration stormpath = new StormpathConfiguration.Builder().baseUrl("https://example.com/").build();
```

Comment:
Different environments are easily achieved by leveraging the gradle build system. The Android gradle plugin generates a `BuildConfig` class at compile time:

```java
/**
 * Automatically generated file. DO NOT MODIFY
 */
package com.stormpath.sdk;

public final class BuildConfig {
  public static final boolean DEBUG = Boolean.parseBoolean("true");
  public static final String APPLICATION_ID = "com.stormpath.sdk";
  public static final String BUILD_TYPE = "debug";
  public static final String FLAVOR = "";
  public static final int VERSION_CODE = -1;
  public static final String VERSION_NAME = "322baf4";
}
```

And there is also a way to define a custom field for this class which is different in each buildType in `build.gradle`:

```groovy
    buildTypes {
        debug {
            buildConfigField 'String', 'API_URL', '"https://api.test.example.com"'
        }
        release {
            minifyEnabled false
            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
            buildConfigField 'String', 'API_URL', '"https://api.example.com"'
        }
    }
```
That way the constant `BuildConfig.API_URL` is generated and we can use it to initialize the Stormpath SDK and the correct one will be used depending on which buildType we are building.

If you think it would help, we can add an example of this in the example app. I don't think we need to do anything in the SDK itself for this.